Thinktecture.Relay.OnPremiseConnector/SignalR/RelayServerConnection.cs

```diff
@@ -309,11 +306,20 @@ private void OnReconnecting()
 			}
 		}
 
-		private Task AcknowlegdeRequest(IOnPremiseTargetRequest request)
+		private Task AcknowlegdeRequestIfRequired(IOnPremiseTargetRequest request)
 		{
-			_logger?.Debug("Sending acknowlegde to relay server. request-id={RequestId}, acknowledge-id={AcknowledgeId}", request.RequestId, request.AcknowledgeId);
+			_logger.Verbose("Checking if acknowledged is required. request-id={RequestId}, acknowledgment-mode={AcknowledgmentMode}, acknowledge-id={AcknowledgeId}", request.RequestId, request.AcknowledgmentMode, request.AcknowledgeId);
+
+			if (request.AcknowledgmentMode == AcknowledgmentMode.Default && !String.IsNullOrEmpty(request.AcknowledgeId))
```

Comment:
Maybe change to a `switch` which does nothing in the other modes, but the other modes are visible? In that case the first check could be for an existing `AcknowledgeId` and early exit when not.